src/stactools/core/utils/convert.py

```diff
@@ -95,26 +95,27 @@ def cogify_subdatasets(
     """
 
     assert_cog_driver_is_enabled()
-    with rasterio.open(infile) as dataset:
-        subdatasets = cast(List[str], dataset.subdatasets)
-        base_file_name = os.path.splitext(os.path.basename(infile))[0]
-        paths = []
-        names = []
-        for subdataset in subdatasets:
-            with rasterio.open(subdataset) as subd:
-                if len(subd.shape) != 2:
-                    continue
-                parts = subdataset.split(":")
-                subdataset_name = parts[-1]
-                if subdataset_names and subdataset_name not in subdataset_names:
-                    continue
-                sanitized_subdataset_name = subdataset_name.replace(" ", "_").replace(
-                    "/", "_"
-                )
-                names.append(sanitized_subdataset_name)
-                file_name = f"{base_file_name}_{sanitized_subdataset_name}.tif"
-                outfile = os.path.join(outdir, file_name)
-                destination_profile = DEFAULT_PROFILE.copy()
-                rasterio.shutil.copy(subdataset, outfile, **destination_profile)
-                paths.append(outfile)
-        return (paths, names)
+    with utils.ignore_not_georeferenced():
```

Comment:
I like having the ignore context utility available via stactools. But I'm shaky on whether to apply it here. IIRC, this warning will get thrown in some cases even when rasterio can find georeferencing information for a subdataset of interest.

However, applying it here will also suppress georeferencing warnings when rasterio can not find subdataset georeferencing information (when you want to see the warning). This is the case for VIIRS, but the warning is suppressed since [georefererencing information is manually being added](https://github.com/stactools-packages/viirs/blob/455ca9600c539999a882f5bfc1f20871103470a0/src/stactools/viirs/cog.py#L115-L116) to the output COGs.

The downside of not applying it here is that a user will have to manually decorate or wrap their call to this function with the new `ignore_not_georeferenced` context manager. But I think it's better to let the user know when something is _potentially_ not georeferenced and allow them to wrap the call with the ignore context if they decide it is not relevant.

Reply:
Ok, yeah that makes sense. I'll remove this usage, and just decorate the test more (this was mostly motivated by me wanting the tests to be quiet 🤫 ).